lib/vnet/setup.go

```diff
@@ -51,62 +63,161 @@ func Run(ctx context.Context, appProvider AppProvider) error {
 		return trace.Wrap(err)
 	}
 
-	return trace.Wrap(manager.Run(ctx))
+	allErrors := make(chan error, 2)
+	g, ctx := errgroup.WithContext(ctx)
+	g.Go(func() error {
+		// Make sure to cancel the context if manager.Run terminates for any reason.
+		defer cancel()
+		err := trace.Wrap(manager.Run(ctx), "running VNet manager")
+		allErrors <- err
+		return err
+	})
+	g.Go(func() error {
+		var adminCommandErr error
+		select {
+		case adminCommandErr = <-adminCommandErrCh:
+			// The admin command exited before the context was canceled, cancel everything and exit.
+			cancel()
+		case <-ctx.Done():
+			// The context has been canceled, the admin command should now exit.
+			adminCommandErr = <-adminCommandErrCh
+		}
+		adminCommandErr = trace.Wrap(adminCommandErr, "running admin subcommand")
+		allErrors <- adminCommandErr
+		return adminCommandErr
+	})
+	// Deliberately ignoring the error from g.Wait() to return an aggregate of all errors.
+	_ = g.Wait()
```

Comment:
unfortunately these PRs are all stacked and such, and rafal is depending on this now, but that's the nature of trying to get a bunch of stuff in at once and waiting for reviews I guess. @ravicious is making a bunch of edits to the way we manage the goroutines/processes in https://github.com/gravitational/teleport/pull/41653 that i think will improve on this and work better for Connect. So if you have any comments on this section they're probably better directed there, unless you see a glaring bug here